crates/bevy_ecs/src/component.rs

```diff
@@ -346,11 +364,34 @@ impl Components {
         self.components.get_unchecked(id.0)
     }
 
+    /// Type-erased equivalent of [`Components::component_id`].
     #[inline]
     pub fn get_id(&self, type_id: TypeId) -> Option<ComponentId> {
         self.indices.get(&type_id).map(|index| ComponentId(*index))
     }
 
+    /// Retrieves the [`ComponentId`] of the given [`Component`] type in
+    /// this [`Components`] instance.
```

Comment:
```suggestion
    /// Retrieves the [`ComponentId`] of the given [`Component`] type.
```

I think “in this `Components` instance” can be safely removed because it is implied by the fact this is a `Components` method.

Reply:
I'd still like to convey that this `ComponentId` is specific to this specific instance of `Components` without the reader drilling down to the `ComponentId` docs. I've replaced it with the following for now:

> The returned `ComponentId` is specific to the `Components` instance it was retrieved from and should not be used with another `Components` instance.